src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (typeof this[i] === 'undefined' || this[i] === null) {
+        this[i] = '';
+      }
+      result += this[i];
+      result += separator;
```

Comment:
You are mutate original array. Do action if `this[i] !== null && this[i] !== undefined
`
```suggestion
      result += `${this[i]}${separator}`;
```

Reply:
done